paasta_tools/cli/cmds/status.py

```diff
@@ -221,10 +268,110 @@ def paasta_status_on_api_endpoint(cluster, service, instance, system_paasta_conf
+
+    if status == kubernetes_tools.KubernetesDeployStatus.Waiting:
+        deploy_status = "%s (new tasks waiting for capacity to become available)" % PaastaColors.red(status_string)
+    elif status == kubernetes_tools.KubernetesDeployStatus.Delayed:
+        deploy_status = "{} (tasks are crashing, next won't launch for another {} seconds)".format(
+                        PaastaColors.red(status_string), backoff_seconds,
+        )
+    elif status == kubernetes_tools.KubernetesDeployStatus.Deploying:
+        deploy_status = PaastaColors.yellow(status_string)
+    elif status == kubernetes_tools.KubernetesDeployStatus.Stopped:
+        deploy_status = PaastaColors.grey(status_string)
+    elif status == kubernetes_tools.KubernetesDeployStatus.Running:
+        deploy_status = PaastaColors.bold(status_string)
+    else:
+        deploy_status = status_string
+
+    return deploy_status
+
+
+def status_kubernetes_job_human(
+    service: str,
+    instance: str,
+    deploy_status: str,
+    desired_app_id: str,
+    app_count: int,
+    running_instances: int,
+    normal_instance_count: int,
+) -> str:
+    name = PaastaColors.cyan(compose_job_id(service, instance))
+
+    if app_count >= 0:
+        if running_instances >= normal_instance_count:
+            status = PaastaColors.green("Healthy")
+            instance_count = PaastaColors.green("(%d/%d)" % (running_instances, normal_instance_count))
+        elif running_instances == 0:
+            status = PaastaColors.yellow("Critical")
+            instance_count = PaastaColors.red("(%d/%d)" % (running_instances, normal_instance_count))
+        else:
+            status = PaastaColors.yellow("Warning")
+            instance_count = PaastaColors.yellow("(%d/%d)" % (running_instances, normal_instance_count))
+        return "Kubernetes:   {} - up with {} instances. Status: {}".format(
+            status, instance_count, deploy_status,
+        )
+    else:
+        status = PaastaColors.yellow("Warning")
+        return "Kubernetes:   {} - {} (app {}) is not configured in Kubernetes yet (waiting for bounce)".format(
+            status, name, desired_app_id,
+        )
+
+
+def print_kubernetes_status(
+    service: str,
+    instance: str,
+    kubernetes_status,
+) -> int:
+    if kubernetes_status.error_message:
+        paasta_print(kubernetes_status.error_message)
+        return 1
+
+    bouncing_status = bouncing_status_human(
+        kubernetes_status.app_count,
+        kubernetes_status.bounce_method,
+    )
+    desired_state = desired_state_human(
+        kubernetes_status.desired_state,
+        kubernetes_status.expected_instance_count,
+    )
+    paasta_print(f"State:      {bouncing_status} - Desired state: {desired_state}")
+
+    status = KubernetesDeployStatus.fromstring(kubernetes_status.deploy_status)
+    if status != KubernetesDeployStatus.NotRunning:
+        if status == KubernetesDeployStatus.Delayed:
```

Comment:
This `Delayed` state isn't defined for k8s right? I wonder if we can find a way to map it to the equivalent thing in k8s...

I'd be tempted to remove the logic otherwise. Anyway, don't consider this a blocker we can leave it in if you don't think it'll do any harm and iterate later.

Reply:
This is actually the second thing I wanted to discuss (beside fixing `-v` for both Marathon/Mesos and k8s setups).
How many states do we need for the k8s setup?
Probably it's possible to deduce `Waiting`/`Delayed`/`NotRunning`/`Stopped`/etc from pods statuses and/or error messages, the question is: do we actually need them?